client/state/comments/selectors/get-post-comments-tree.js

```diff
@@ -28,6 +28,17 @@ export const getPostCommentsTree = treeSelect(
 			) {
 				return false;
 			}
+
+			// Filter out pending comments that are duplicates of approved comments
+			if ( item.status === 'pending' ) {
+				const duplicate = allItems.find(
+					( comment ) => comment.content === item.content && comment.ID !== item.ID
+				);
+				if ( duplicate ) {
+					return false;
+				}
+			}
```

Comment:
I think this solution has some downsides - what happens if someone submits a pending comment that has the same content as an existing approved comment? 

I think we have to look for the original reason for the comment to get duplicated and fix it there. It could be something we're doing twice in the reducer, or it could be two conflicting actions - one that adds the comment optimistically and then one that adds it after it has actually been submitted. Maybe those two are supposed to be compared and the second one would be inserted only if the comparison fails, and the comparison has been failing for some time? 

I'd suggest diving deeper and finding the root cause, instead of patching it up and concealing it under yet another level of complexity. 

I'd also encourage writing some tests. This is a great use case for some component tests that ensure this is being fixed, and won't happen again. I'm happy to provide on how to write component tests with `@testing-library/react` if that is going to help.

Reply:
ok, I will keep digging 👍